flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -77,19 +85,20 @@
 
 	private ViewUpdater viewUpdater;
 
-	private final ScopeFormats scopeFormats;
-	private final char globalDelimiter;
-	private final List<Character> delimiters = new ArrayList<>();
+	private boolean isShutdown;
 
 	/**
 	 * Creates a new MetricRegistry and starts the configured reporter.
 	 */
 	public MetricRegistryImpl(MetricRegistryConfiguration config) {
 		this.scopeFormats = config.getScopeFormats();
 		this.globalDelimiter = config.getDelimiter();
+		this.delimiters = new ArrayList<>(10);
```

Comment:
the number of delimiters is always equal to the number of reporters, so they should be initialized with the same size.

Reply:
Good point, will change it.